backend/external/imagebb.js

```diff
@@ -16,18 +30,19 @@ const uploadImage = async (base64) => {
 
   const form = new FormData();
   form.append('image', imageData);
-  const url = `${api}?expiration=600&key=${API_KEY}`;
-
   const response = await fetch(url, {
     method: 'POST',
+    headers: headers,
     body: form
   });
   const imageResponse = await response.json();
-
   try {
-    return {
+    return Hosting == 'imgbb' ?  {
       imageurl: imageResponse.data.image.url
-    };
+    } 
+      :  {
+        imageurl: imageResponse.data.link
+      }
```

Comment:
You might want to write the response handler separately in their respective lib file (for imgbb its `backend/external/imagebb.js` ) and import at the top based on the chosen provider.